File: dotnet/src/Connection/Create.cs
namespace Pong
{
  partial class Connection
  {
    /// <summary>
    /// A message that is sent when a session is created
    /// <br/>
    /// Sent to all players who are not in a session (but including the two players in the new session)
    /// </summary>
    public class CreateEvent : GameEvent
    {
      public string Player1 { get; set; }
      public string Player2 { get; set; }

      public CreateEvent(string player1, string player2)
        : base("create")
      {
        Player1 = player1;
        Player2 = player2;
      }

      /// <summary>
      /// Deserialize a CreateEvent from json
      /// </summary>
      public static new CreateEvent Deserialize(string json)
      {
        return Deserialize<CreateEvent>(json);
      }
    }
  }
}

File: server/src/messages/create.ts
import Player from "@/players/player";

/**
 * A message that is sent when a session is created
 *
 * Sent to all players who are not in a session (but including the two players in the new session)
 */
export interface CreateMessage {
  type: "create";
  player1: string;
  player2: string;
}

/**
 * Inform everyone that a session has been created between two players
 *
 * Sent to all players who are not in a session (but including the two players in the new session)
 * @param {Player} player1 The player who sent the Create
 * @param {Player} player2 The player who received the Create
 * @returns {CreateMessage} An Create message
 */
function Create(player1: Player, player2: Player): CreateMessage {
  return {
    type: "create",
    player1: player1.name,
    player2: player2.name,
  };
}

export default Create;
